### PR TITLE
Mark PDB files generated by R2R as ExcludeFromSingleFile

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
When published with `PublishReadyToRun=true` and `PublishReadyToRunEmitSymbols=true` certain additional symbol files are generated:
On Windows: `<app>.ni.pdb`
On Linux: `<app>.<guid>.map`

When additionally publishing with `PublishSingleFile=true` exclude the above files from being bundled into the single-file output.

### CODE REVIEW

**Comment by swaroop-sridhar** [comment]
CC: @fadimounir  @nguerrera

**Comment by swaroop-sridhar** [comment]
The darwin failures seem unrelated to this change (and are found in other PRs too).

**Comment by swaroop-sridhar** [comment]
CC: @sbomer

**Comment by swaroop-sridhar** [comment]
@nguerrera  please review.

File: src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs [hunk 1]
```diff
@@ -184,6 +186,11 @@ void ProcessInputFileList(ITaskItem[] inputFiles, List<ITaskItem> imageCompilati
                         r2rSymbolsFileToPublish.ItemSpec = outputPDBImage;
                         r2rSymbolsFileToPublish.SetMetadata(MetadataKeys.RelativePath, outputPDBImageRelativePath);
                         r2rSymbolsFileToPublish.RemoveMetadata(MetadataKeys.OriginalItemSpec);
+                        if(!IncludeSymbolsInSingleFile)
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs]: nit: missing space
```suggestion
                        if (!IncludeSymbolsInSingleFile)
```

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs [hunk 2]
```diff
@@ -142,6 +144,36 @@ public void It_generates_a_single_file_including_pdbs()
                 .OnlyHaveFiles(expectedFiles);
         }
 
+        [WindowsOnlyFact]
```
Comment (nguerrera) [inline src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs]: Why is this WindowsOnly?

**Review by nguerrera — APPROVED** [review]

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs (hunk 2 above)
Comment (swaroop-sridhar) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs]: The `ni.pdb` files are only generated on Windows.

Comment (nguerrera) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs]: ah, of course.

**Review by nguerrera — APPROVED** [review]

**Pull request merged**